pkg/controller/metric.go

```diff
@@ -27,9 +27,19 @@ var (
 		},
 		[]string{"event"},
 	)
+	lastReconciledGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: PrometheusNamespace,
+			Subsystem: PrometheusSubsystem,
+			Name:      "last_reconciled",
+			Help:      "",
+		},
+		[]string{"kind", "name", "namespace"},
```

Comment:
This is another proposition to have metric with an unbound label cardinality. This disturbs me.

But if @giantswarm/sig-monitoring and @giantswarm/team-atlas are ok with that I'm fine too.

Reply:
Thanks @kopiczko good to raise this. We're getting more CRs in the control planes. So this could be an issue in busy CPs with lots of TCs. I guess app CRs and certconfig CRs are the most numerous.

@tomahawk28 Let's be sure Atlas and the SIG are fine with this.

Reply:
This would create lots of metrics labels by considering the number of apps CRs/chart CRs in each CP. (appx. 60 for each)
I would like to know whether we have any limitations or guidelines limiting the number of label values. 

One thing might we can considering is that, turn these alerts off if we use operatorkit for general k8s objects, likely secret, service objects. Since this could be an hour-long discussion, I would move it to the sig-monitoring channel.

Reply:
Could we get away with just having the kind? So, we'd have one metric per operator / controller. I can't personally think of a situation where we'd only reconcile _some_ CRs

Reply:
Thanks @JosephSalisbury,  In the case of `App`, `Charts` CRD, and some provider-specific CRDs, they are reconciled to the operators with matching versions. Because of this, only `kind` value is not enough to see the reconciliation problem. 
do we have any controller-specific value which we could identify the matching labels? likely selector?

Reply:
I don't think I follow why only having the `kind` label isn't sufficient - from https://github.com/giantswarm/operatorkit/pull/501#issuecomment-753994356, it looks like all custom resources have very similar timestamps.

Reply:
I thought that we need a fingerprint for each controller, for example, cluster-operator-0-23-16 and cluster-operator-0-23-17 would be reconciled to the same kind type but totally different matching labels, right?

See belows for labels. 
```
✔{app="app-operator",app_instance="app-operator-unique",app_name="app-operator",app_version="0.0.0",cluster_id="ginger",cluster_type="host",exported_namespace="giantswarm",instance="100.64.5.144:8000",job="host-cluster-ginger-workload",kind="App",name="vertical-pod-autoscaler-app-unique",namespace="giantswarm",node="ip-10-0-5-53.eu-west-1.compute.internal",pod_phase="Running",pod_ready="true",provider="aws"}

```
But now I also realized maybe we could utilize `app_version` and `app_name` to let us know the location of operators. 
so having a `kind` label only is also an option, WDYT?

Reply:
yeah, different operators will have different `instance` labels, so they'll be differentiable. if you're happy with it, i think it lets us sidestep the whole unbounded cardinality issue

Reply:
From the title of https://github.com/giantswarm/giantswarm/issues/14847 it sounds to me like the only labels that are really needed are:

- operator name
- operator version
- controller name

Based on that you can clearly tell which controller in which operator in which version is not reconciling.

Reply:
Sure, I have `name`, `app_version`, and `kind` as labels, I don't think the controller name is worth it since `kind` itself already represents such information.

Reply:
There can be a situation in which 2 separate controllers reconcile the same kind. Maybe it isn't very useful but possible. BTW that's going to happen in config-controller (as long as we don't have AppCatalogEntry CRs for test catalogs).

Reply:
If the case that multiple controllers are reconciled into the same kind of object is prevalent, then I think adding the controller name into the label makes sense. thanks to let me know that.

Reply:
I tested with an additional label with controller name, but found out most of the cases, we are assinging `project.Name()` to the name of the controller. 
https://github.com/giantswarm/app-operator/blob/5e081b766dbee2e14cb3f5869946f97302f0b9ea/service/controller/app/app.go#L98-L115
@kopiczko I think that's something we should modify from the operator itself, not operatorkit should do, WDYT?
PTAL.

Reply:
> but found out most of the cases, we are assinging `project.Name()` to the name of the controller.

That's ok.

Reply:
I'd still remove kind label but I won't block because of that. I think what controller is watching is irrelevant from the alert standpoint.

Reply:
As we have a controller name as label, so now I agree we don't need kind information. I will delete that we can introduce later if we really need that.